File: packages/rust-sdk/src/operations/analytics.rs
use crate::client::StraitClient;
use crate::errors::StraitError;
use std::sync::Arc;

pub struct AnalyticsService {
    client: Arc<StraitClient>,
}

impl AnalyticsService {
    pub fn new(client: Arc<StraitClient>) -> Self {
        Self { client }
    }

    pub async fn get_performance(&self, query: Option<&[(&str, &str)]>) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/analytics/performance", query, None, None).await
    }
}

File: packages/go-sdk/operations/analytics.go
package operations

import "context"

// AnalyticsService provides performance analytics operations.
type AnalyticsService struct{ r Requester }

func NewAnalyticsService(r Requester) *AnalyticsService { return &AnalyticsService{r: r} }

func (s *AnalyticsService) GetPerformance(ctx context.Context, query map[string]string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/v1/analytics/performance", query, nil, nil, &result)
	return result, err
}
